### PR TITLE
Trusted nets

### PR DESCRIPTION
**Pull request by ux-AylonS**

### CODE REVIEW

File: src/components/RbzTable.vue [hunk 1]
```diff
@@ -216,10 +227,13 @@ export default defineComponent({
     showNewButton: Boolean,
     rowClickable: Boolean,
     showRowButton: Boolean,
-    showSecondRowButton: Boolean,
+    showSecondButtonRow: Boolean,
```
Comment (Aviv-Galmidi) [inline src/components/RbzTable.vue:230]: That's incorrect
We're showing a second row-button (button of a row) not a second button row (row of buttons)

File: src/doc-editors/ProxyTemplatesEditor.vue [hunk 2]
```diff
@@ -446,7 +446,7 @@
           </div>
         </div>
       </div>
-      <!--div-- class="card collapsible-card"
+      <div class="card collapsible-card"
              :class="{ collapsed: isTrustedCollapsed }">
```
Comment (Aviv-Galmidi) [inline src/doc-editors/ProxyTemplatesEditor.vue:450]: `isTrustedCollapsed` -> `isTrustedSourcesCollapsed`

File: src/doc-editors/ProxyTemplatesEditor.vue [hunk 3]
```diff
@@ -469,20 +469,39 @@
...
                                 >
                     </rbz-table>
                 </div>
             </div>
+            <div class="trusted-modal-input-container ml-30"
+                 v-if="showEditTrustedSource || isAddModalVisible" >
+              <input type="text"
+                    class="ip-input is-size-7 ellipsis"
+                    v-model="newAddress"
+              />
```
Comment (Aviv-Galmidi) [inline src/doc-editors/ProxyTemplatesEditor.vue:491]: Fix spacing and indents -
`/>` closing should be in the same line as the last property
different properties should be aligned vertically (`type`, `class`, etc.)

File: src/doc-editors/ProxyTemplatesEditor.vue [hunk 4]
```diff
@@ -469,20 +469,39 @@
...
+                                :show-second-button-row="true"
                                 @row-button-clicked="deleteTrustedElement"
                                 @new-button-clicked="toggleAddingNewTrustedSource"
-                                @second-row-button-clicked="toggleEditTrustedElement"
+                                @second-button-row-clicked="toggleEditTrustedElement"
                                 >
                     </rbz-table>
                 </div>
             </div>
+            <div class="trusted-modal-input-container ml-30"
```
Comment (Aviv-Galmidi) [inline src/doc-editors/ProxyTemplatesEditor.vue:486]: I don't believe `ml-30` exists in bulma, `ml` usually stands for margin-left and in bulma they go from 0 to 6

File: src/doc-editors/ProxyTemplatesEditor.vue [hunk 5]
```diff
@@ -899,128 +920,112 @@ export default defineComponent({
...
+      const id = this.trustedData.length
+      const newTrustedElement = {id: id, address: this.newAddress, comment: this.newComment}
+      this.trustedData.push(newTrustedElement)
+      this.isAddModalVisible=false
+    },
 
-    //   this.currentEditIndex = this.trustedData.findIndex((trusted) => trusted.id ===id)
-    //   this.editAddress = this.trustedData[this.currentEditIndex].address
-    //   this.editComment = this.trustedData[this.currentEditIndex].comment
+    // openEditTrustedSource(idx: number) {
```
Comment (Aviv-Galmidi) [inline src/doc-editors/ProxyTemplatesEditor.vue:951]: commented out on purpose?

File: src/doc-editors/ProxyTemplatesEditor.vue [hunk 6]
```diff
@@ -899,128 +920,112 @@ export default defineComponent({
...
+      this.trustedData = trustedArr.filter((trusted) => trusted.id !== id)
+      const dataTrusted = this.trustedData.map((trusted) => {
+        return {address: trusted.address, comment: trusted.comment}
+      })
+      const data = {
+        id: this.planetID,
+        name: this.planetName,
+        trusted_nets: dataTrusted,
+      }
+      console.log('delete id', id, 'data', data)
```
Comment (Aviv-Galmidi) [inline src/doc-editors/ProxyTemplatesEditor.vue:1024]: Let's remove the console logs we don't need

File: src/doc-editors/ProxyTemplatesEditor.vue [hunk 7]
```diff
@@ -1029,4 +1034,30 @@ export default defineComponent({
...
+
+  .trusted-modal-input-container {
+    display: flex;
+    height: 55px;
+    margin: 10px 20px;
+  }
+
+  .ip-input {
+    height: 40px;
+    width: 47%;
```
Comment (Aviv-Galmidi) [inline src/doc-editors/ProxyTemplatesEditor.vue:1046]: `width: 47%` seems like a magic number, it's very specific
Can we avoid it?

File: src/doc-editors/ProxyTemplatesEditor.vue [hunk 8]
```diff
@@ -1029,4 +1034,30 @@ export default defineComponent({
...
+
+  .comment-input {
+    height: 40px;
+    width: 47%;
+  }
+
+  .submit-changes {
+    width: 30px;
+
+    button {
```
Comment (Aviv-Galmidi) [inline src/doc-editors/ProxyTemplatesEditor.vue:1057]: Let's avoid editing `button` directly and give these buttons a class

File: src/doc-editors/ProxyTemplatesEditor.vue [hunk 9]
```diff
@@ -610,46 +633,51 @@ export default defineComponent({
...
-      // tagRule: '',
-      // newAddress: '127.0.0.0/8',
-      // newComment: 'Private subnet',
-      // editAddress: '',
-      // editComment: '',
-      // isEdit: false,
-      // entryType: 'cidr',
-      // isError: false,
-      // currentEditIndex: 0,
+      planetID: null,
```
Comment (Aviv-Galmidi) [inline src/doc-editors/ProxyTemplatesEditor.vue:636]: This seems like an exessive list of new properties, and we don't use them all - for exmaple there is no call for `redis` other than this definition
Let's clean it up

File: src/doc-editors/ProxyTemplatesEditor.vue [hunk 10]
```diff
@@ -469,20 +469,39 @@
...
+              <input type="text"
+                    class="ip-input is-size-7 ellipsis"
+                    v-model="newAddress"
+              />
+              <input  type="text"
+                      class="comment-input is-size-7 ellipsis"
+                      v-model="newComment"
+              />
+              <div class="submit-changes" v-if="isAddModalVisible">
+                <button class="btn" @click="addNewTrustedSource" ><i class="fas fa-plus"></i></button>
```
Comment (Aviv-Galmidi) [inline src/doc-editors/ProxyTemplatesEditor.vue:497]: Use bulma button classes

File: src/doc-editors/ProxyTemplatesEditor.vue [hunk 11]
```diff
@@ -469,20 +469,39 @@
...
+              </div>
+              <div class="submit-changes" v-if="showEditTrustedSource">
+                <button class="btn" @click="editTrustedSource" ><i class="fas fa-edit"></i></button>
```
Comment (Aviv-Galmidi) [inline src/doc-editors/ProxyTemplatesEditor.vue:500]: Use bulma button classes

**Review by Aviv-Galmidi — CHANGES_REQUESTED** [review]

File: src/components/RbzTable.vue (hunk 1 above)
Comment (ux-AylonS) [reply src/components/RbzTable.vue:230]: well, a second button for the row
it depends where you put the comma
second button, row
second, row-button
second , button-row.

File: src/doc-editors/ProxyTemplatesEditor.vue (hunk 7 above)
Comment (ux-AylonS) [reply src/doc-editors/ProxyTemplatesEditor.vue:1046]: it is specific for macbook screen size

Comment (ux-AylonS) [reply src/doc-editors/ProxyTemplatesEditor.vue:1046]: what would you suggest?

File: src/doc-editors/ProxyTemplatesEditor.vue [hunk 12]
```diff
@@ -1043,21 +1033,23 @@ export default defineComponent({
...
   .ip-input {
     height: 40px;
-    width: 47%;
+    width: 45%;
   }
 
   .comment-input {
     height: 40px;
-    width: 47%;
+    width: 46%;
```
Comment (Aviv-Galmidi) [inline src/doc-editors/ProxyTemplatesEditor.vue:1041]: Why exactly `46%` it's very specfic, what's the use case of these inputs? Can we use css flex / bulma columns?

**Review by Aviv-Galmidi — CHANGES_REQUESTED** [review]

File: src/doc-editors/ProxyTemplatesEditor.vue [hunk 13]
```diff
@@ -486,10 +486,10 @@
             <div class="trusted-modal-input-container"
                  v-if="showEditTrustedSource || isAddModalVisible" >
               <input type="text"
-                     class="ip-input is-size-7 ellipsis"
+                     class="input is-small ip-input data-cell is-size-7 data-cell-content ellipsis"
```
Comment (Aviv-Galmidi) [inline src/doc-editors/ProxyTemplatesEditor.vue:489]: `data-cell` and `data-cell-content` are RbzTable classes, they won't do anything here

**Comment by Aviv-Galmidi** [comment]
Moved to #135

**Pull request closed without merging**